http/src/main/scala/org/broadinstitute/dsde/workbench/leonardo/api/RuntimeRoutes.scala

```diff
@@ -0,0 +1,145 @@
+package org.broadinstitute.dsde.workbench.leonardo
+package http
+package api
+
+import _root_.java.time.Instant
+import java.util.UUID
+import java.util.concurrent.TimeUnit
+
+import akka.http.scaladsl.marshalling.ToResponseMarshallable
+import akka.http.scaladsl.model.StatusCodes
+import akka.http.scaladsl.server
+import akka.http.scaladsl.server.Directives._
+import cats.effect.{IO, Timer}
+import cats.mtl.ApplicativeAsk
+import io.circe.Decoder
+import org.broadinstitute.dsde.workbench.leonardo.http.api.LeoRoutes.validateRuntimeNameDirective
+import org.broadinstitute.dsde.workbench.leonardo.http.service.{RuntimeConfigRequest, RuntimeService}
+import org.broadinstitute.dsde.workbench.model.{TraceId, UserInfo}
+import org.broadinstitute.dsde.workbench.model.google.{GcsPath, GoogleProject}
+import RuntimeRoutes._
+import de.heikoseeberger.akkahttpcirce.ErrorAccumulatingCirceSupport._
+import LeoRoutesJsonCodec.dataprocConfigDecoder
+import JsonCodec._
+import scala.concurrent.duration._
+
+class RuntimeRoutes(runtimeService: RuntimeService[IO], userInfoDirectives: UserInfoDirectives)(
+  implicit timer: Timer[IO]
+) {
+  val routes: server.Route = userInfoDirectives.requireUserInfo { userInfo =>
+    implicit val traceId = ApplicativeAsk.const[IO, TraceId](TraceId(UUID.randomUUID()))
+    // https://doc.akka.io//docs/akka-http/current/routing-dsl/path-matchers.html?language=scala#examples
+    pathPrefix("google" / "v1" / "runtime") {
+      pathPrefix(googleProjectSegment / Segment) { (googleProject, runtimeNameString) =>
+        validateRuntimeNameDirective(runtimeNameString) { runtimeName =>
+          pathEndOrSingleSlash {
+            post {
+              entity(as[CreateRuntime2Request]) { req =>
+                complete(
+                  createRuntimeHandler(
+                    userInfo,
+                    googleProject,
+                    runtimeName,
+                    req
+                  )
+                )
+              }
+            }
+          }
+        }
+      }
+    }
+  }
+
+  private[api] def createRuntimeHandler(userInfo: UserInfo,
+                                        googleProject: GoogleProject,
+                                        runtimeName: RuntimeName, //TODO: rename this to RuntimeName once Rob's PR is in
+                                        req: CreateRuntime2Request): IO[ToResponseMarshallable] =
```

Comment:
Should this return some specific response type instead of `ToResponseMarshallable`?

Reply:
I see this function as the helper that fits into `complete(???)`...so I think it should be `ToResponseMarshallable`, at "service" level, each function is returning api specific responses